code/modules/projectiles/guns/projectile/launchers.dm

```diff
@@ -83,3 +83,752 @@
+			for(var/i in fire_stop)
+				if(istype(walker, i))
+					set_fire = FALSE
+					break
+				if(locate(i) in walker.contents)
+					set_fire = FALSE
+					break
+			if(!set_fire)
+				break
+			for(var/obj/O in walker.contents)
+				if(istype(O, /obj/structure/window))
+					var/obj/structure/window/W = O
+					W.take_damage(W.obj_integrity)
+				if(istype(O, /obj/structure/grille))
+					var/obj/structure/window/W = O
+					W.take_damage(W.obj_integrity)
+				if(istype(O, /obj/machinery/door))
+					if(O.density)
+						set_fire = FALSE
+						break
+
+		if(set_fire)
+			new /obj/effect/hotspot(T)
+			ffx_hit(T)
+			for(var/mob/living/L in T)
+				L.EyeBlurry(10)
+
+	exsfx_hit(ex_range+4, ex_range+10, 'sound/effects/explosion1.ogg')
+	var/datum/effect_system/explosion/smoke/E = new/datum/effect_system/explosion/smoke()
+	E.set_up(target)
+	E.start()
+	return TRUE
+
+/obj/item/ammo_casing/rocket/a66mm
+	name = "66mm HE rocket shell"
+	desc = "A high explosive rocket designed to be fired from a launcher."
+	icon_state = "rocketshell"
+	projectile_type = /obj/item/projectile/a66mm
+	caliber = "rocket"
+	
+/obj/item/ammo_casing/rocket/a66mm/rod
+	name = "66mm Rod Round rocket shell"
+	desc = "A rod rocket designed to be fired from a launcher."
+	icon_state = "rocketshell"
+	projectile_type = /obj/item/projectile/a66mm/rod
+	caliber = "rocket"
+	
+/obj/item/ammo_casing/rocket/a66mm/frag
+	name = "66mm Fragmentation rocket shell"
+	desc = "A fragmentation rocket designed to be fired from a launcher."
+	icon_state = "rocketshell"
+	projectile_type = /obj/item/projectile/a66mm/frag
+	caliber = "rocket"
+	
+/obj/item/ammo_casing/rocket/a66mm/emp
+	name = "66mm EMP rocket shell"
+	desc = "A emp rocket designed to be fired from a launcher."
+	icon_state = "rocketshell"
+	projectile_type = /obj/item/projectile/a66mm/emp
+	caliber = "rocket"
+	
+/obj/item/ammo_casing/rocket/a66mm/incen
+	name = "66mm Incendiary rocket shell"
+	desc = "A incendiary rocket designed to be fired from a launcher."
+	icon_state = "rocketshell"
+	projectile_type = /obj/item/projectile/a66mm/incen
+	caliber = "rocket"
+	
+/obj/item/ammo_box/magazine/internal/S72LRW
+	name = "rocket launcher internal magazine"
+	ammo_type = /obj/item/ammo_casing/rocket/a66mm
+	caliber = "66mm"
+	max_ammo = 1
+	
+/obj/item/ammo_box/magazine/internal/S72LRW/rod
+	ammo_type = /obj/item/ammo_casing/rocket/a66mm/rod
+	
+/obj/item/ammo_box/magazine/internal/S72LRW/frag
+	ammo_type = /obj/item/ammo_casing/rocket/a66mm/frag
+	
+/obj/item/ammo_box/magazine/internal/S72LRW/emp
+	ammo_type = /obj/item/ammo_casing/rocket/a66mm/emp
+	
+/obj/item/ammo_box/magazine/internal/S72LRW/incen
+	ammo_type = /obj/item/ammo_casing/rocket/a66mm/incen
+
+/obj/item/projectile/a66mm/proc/ffx_hit(atom/A)
+	if(!A)
+		return
+	var/turf/T = get_turf(A)
+	if(isfloorturf(T))
+		var/turf/simulated/floor/F = T
+		F.break_tile()
+		F.burn_tile()
+		
+/obj/item/projectile/a66mm/proc/friendly_fire_dud(turf/target as turf, in_range)
```

Comment:
Again bit odd design wise. Will also make the gun useless for the crew to use once they kill the nukie

Reply:
Also remove the `as turf`

Reply:
The friendly fire feature can be removed if need be.